src/fheroes2/dialog/dialog_graphics_settings.cpp

```diff
@@ -0,0 +1,224 @@
+        const fheroes2::Point shadowOffset( dialogOffset.x - BORDERWIDTH, dialogOffset.y );
+
+        const fheroes2::Rect windowRoi{ dialogOffset.x, dialogOffset.y, dialog.width(), dialog.height() };
+
+        const fheroes2::ImageRestorer restorer( display, shadowOffset.x, shadowOffset.y, dialog.width() + BORDERWIDTH, dialog.height() + BORDERWIDTH );
+
+        fheroes2::Blit( dialogShadow, display, windowRoi.x - BORDERWIDTH, windowRoi.y + BORDERWIDTH );
+        fheroes2::Blit( dialog, display, windowRoi.x, windowRoi.y );
+
+        const fheroes2::Rect windowResolutionRoi( resolutionRoi + windowRoi.getPosition() );
+        const fheroes2::Rect windowModeRoi( modeRoi + windowRoi.getPosition() );
+        const fheroes2::Rect windowVSyncRoi( vSyncRoi + windowRoi.getPosition() );
+        const fheroes2::Rect windowSystemInfoRoi( systemInfoRoi + windowRoi.getPosition() );
+
+        drawResolution( windowResolutionRoi );
+        drawMode( windowModeRoi );
+        drawVSync( windowVSyncRoi );
+        drawSystemInfo( windowSystemInfoRoi );
+
+        const fheroes2::Point buttonOffset( 112 + windowRoi.x, 252 + windowRoi.y );
+        fheroes2::Button okayButton( buttonOffset.x, buttonOffset.y, isEvilInterface ? ICN::SPANBTNE : ICN::SPANBTN, 0, 1 );
+        okayButton.draw();
+
+        display.render();
+
+        LocalEvent & le = LocalEvent::Get();
+        while ( le.HandleEvents() ) {
+            if ( le.MousePressLeft( okayButton.area() ) ) {
+                okayButton.drawOnPress();
+            }
+            else {
+                okayButton.drawOnRelease();
+            }
+
+            if ( le.MouseClickLeft( okayButton.area() ) || Game::HotKeyCloseWindow() ) {
+                break;
+            }
+            if ( le.MouseClickLeft( windowResolutionRoi ) ) {
+                return SelectedWindow::Resolution;
+            }
+            if ( le.MouseClickLeft( windowModeRoi ) ) {
+                return SelectedWindow::Mode;
+            }
+            if ( le.MouseClickLeft( windowVSyncRoi ) ) {
+                return SelectedWindow::VSync;
+            }
+            if ( le.MouseClickLeft( windowSystemInfoRoi ) ) {
+                return SelectedWindow::SystemInfo;
+            }
+
+            if ( le.MousePressRight( windowResolutionRoi ) ) {
+                fheroes2::showStandardTextMessage( _( "Select Game Resolution" ), _( "Change the resolution of the game." ), 0 );
+            }
+            else if ( le.MousePressRight( windowModeRoi ) ) {
+                fheroes2::showStandardTextMessage( _( "window|Mode" ), _( "Toggle between fullscreen and windowed modes." ), 0 );
+            }
+            else if ( le.MousePressRight( windowVSyncRoi ) ) {
+                fheroes2::showStandardTextMessage( _( "V-Sync" ), _( "Enabled V-Sync option is useful for some monitors to resolve flickering issues." ), 0 );
```

Comment:
What do you think about phrasing it like this? I'm not sure about the line length of the actual code here and if the code style check will fail.

```suggestion
                fheroes2::showStandardTextMessage( _( "V-Sync" ), _( "The V-Sync option can be enabled  to resolve flickering issues on some monitors." ), 0 );
```

Reply:
Yup, looks better.